### ISSUE TITLE
IPv6 and TLS

### ISSUE THREAD

**Issue by verticelo** [irontec/sngrep#255]
Hi there! First of all, thanks for SNGREP! What a great tool!

I can see that IPv6 is support and TLS is supported. However, is the combination of them supported? I only get IPv6/UDP and IPv4/TLS working. Not the combination of both. I recorded some and decrypted it fine in Wireshark.

If they are supported, then I'm happy to do some debugging if any pointers can be given. Cheers!

**Comment by Kaian** [comment]
Hi @verticelo 

Glad to hear it helped you! 

We dont use TLS as much as we should, so it may probably fail in some scenarios. 

Sngrep only supports a couple of ciphers, but if wireshark can decrypt it we can give it a try.

If you can generate some test data (a .pcap and a private keyfile) and send it to my email I can try to debug where the problem is. 

Thanks in advance!

**Comment by verticelo** [comment]
I do not believe it's a problem with ciphers, since the same cipher works over IPv4 with sngrep. It's only when using IPv6 it doesn't work.

Have you ever tried IPv6/TLS and seen that it works?

I will generate some debug data next week. Thanks!

**Comment by Kaian** [comment]
> Have you ever tried IPv6/TLS and seen that it works?

Nope. We rarely use the decrypting functions of sngrep. Almost all testing and code has been done by community requests and with community tests data. I may be wrong, but I think none of those test files where IPv6

> I will generate some debug data next week. Thanks!

Thanks!!

**Comment by verticelo** [comment]
Hi again,

Sorry for the late reply. I have now generated pcaps and certificates in our dev environment and sent them to you by email. The pcap is decryptable by Wireshark but not by sngrep.

Have a great day!
